torch/multiprocessing/reductions.py

```diff
@@ -67,12 +67,12 @@ def free_dead_references(self):
 shared_cache = SharedCache()
 
 
-def rebuild_event(handle):
-    return torch.cuda.Event(_handle=handle)
+def rebuild_event(device, handle):
+    return torch.cuda.Event.from_ipc_handle(device, handle)
 
 
 def reduce_event(event):
-    return (rebuild_event, (event.ipc_handle(),))
+    return (rebuild_event, (torch.cuda.current_device(), event.ipc_handle()))
```

Comment:
This should be the event's device, not the current device. You'll also want to call `event.ipc_handle()` before accessing the event's device to ensure that the event is created (and has a device).